### PR TITLE
feat: reduce ledger polling on heavy duty work like recover accounts

### PR DESCRIPTION
**Pull request by begonaalvarezd**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

Close https://github.com/iotaledger/firefly/issues/7714

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/account/actions/findBalances.ts [hunk 1]
```diff
@@ -16,11 +18,13 @@ let searchAccountStartIndex: number
...
 let _addressGapLimit: number
 
+let profileType: ProfileType
+
 export function initialiseAccountRecoveryConfiguration(
     algortihmType: SearchAlgorithmType,
     config?: RecoverAccountsPayload
 ): void {
-    const profileType = get(activeProfile)?.type
+    profileType = get(activeProfile)?.type
```
Comment (marc2332) [inline packages/shared/lib/core/account/actions/findBalances.ts:27]: Let's try not to use global functions to store state between functions. If you want to get the profile Type in `findBalances` just call `get` on the store again.

**Review by marc2332 — CHANGES_REQUESTED** [review]

**Review by cpl121 — DISMISSED** [review]
Works, but I think it would be interesting to also add it to the shimmer claiming

File: packages/shared/lib/core/ledger/actions/pollLedgerNanoStatus.ts [hunk 2]
```diff
@@ -1,28 +1,36 @@
...
+
+        void pollingFunction()
     }
 }
 
 export function stopPollingLedgerNanoStatus(): void {
     if (get(isPollingLedgerDeviceStatus)) {
-        clearInterval(intervalTimer)
+        if (intervalTimer) {
+            clearInterval(intervalTimer)
```
Comment (marc2332) [inline packages/shared/lib/core/ledger/actions/pollLedgerNanoStatus.ts:32]:
```suggestion
        if (timeoutTimer) {
            clearTimeout(intervalTimer)
```

File: packages/shared/lib/core/ledger/actions/pollLedgerNanoStatus.ts [hunk 3]
```diff
@@ -1,28 +1,36 @@
...
-        void getAndUpdateLedgerNanoStatus(profileManager)
-        intervalTimer = setInterval(() => {
-            void getAndUpdateLedgerNanoStatus(profileManager)
-        }, pollInterval)
         isPollingLedgerDeviceStatus.set(true)
+        const pollingFunction = async (): Promise<void> => {
+            await getAndUpdateLedgerNanoStatus(profileManager)
+            const isLedgerBusy = get(ledgerNanoStatus)?.busy
+            const currentPollInterval = isLedgerBusy ? slowedPollInterval : defaultPollInterval
+            intervalTimer = setTimeout(() => void pollingFunction(), currentPollInterval)
```
Comment (marc2332) [inline packages/shared/lib/core/ledger/actions/pollLedgerNanoStatus.ts:22]:
```suggestion
            timeoutTimer = setTimeout(() => void pollingFunction(), currentPollInterval)
```

File: packages/shared/lib/core/ledger/actions/pollLedgerNanoStatus.ts [hunk 4]
```diff
@@ -1,28 +1,36 @@
...
+import { DEFAULT_LEDGER_NANO_STATUS_POLL_INTERVAL } from '../constants'
 import { deconstructLedgerNanoStatusPollingConfiguration } from '../helpers'
 import { ILedgerNanoStatusPollingConfiguration } from '../interfaces'
-import { isPollingLedgerDeviceStatus } from '../stores'
-
+import { isPollingLedgerDeviceStatus, ledgerNanoStatus } from '../stores'
 import { getAndUpdateLedgerNanoStatus } from './getAndUpdateLedgerNanoStatus'
 
-let intervalTimer: ReturnType<typeof setInterval>
+let intervalTimer: ReturnType<typeof setInterval> | null
```
Comment (marc2332) [inline packages/shared/lib/core/ledger/actions/pollLedgerNanoStatus.ts:8]:
```suggestion
let timeoutTimer: ReturnType<typeof setTimeout> | null
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: packages/shared/lib/core/ledger/actions/pollLedgerNanoStatus.ts [hunk 5]
```diff
@@ -5,7 +5,7 @@ import { ILedgerNanoStatusPollingConfiguration } from '../interfaces'
 import { isPollingLedgerDeviceStatus, ledgerNanoStatus } from '../stores'
 import { getAndUpdateLedgerNanoStatus } from './getAndUpdateLedgerNanoStatus'
 
-let intervalTimer: ReturnType<typeof setInterval> | null
+let timeoutTimer: ReturnType<typeof setInterval> | undefined
```
Comment (marc2332) [inline packages/shared/lib/core/ledger/actions/pollLedgerNanoStatus.ts:8]:
```suggestion
let timeoutTimer: ReturnType<typeof setTimeout> | undefined
```

**Review by marc2332 — CHANGES_REQUESTED** [review]
sorry

**Pull request merged**